### ISSUE TITLE
How do I start a bloodhound search when typeahead:selected fires

### ISSUE THREAD

**Issue by bryanrasmussen** [corejavascript/typeahead.js#146]
Ok this has been driving me nuts all day. 
When a user selects from the drop down list a value I want a bloodhound search to be fired. It seems that the default behavior is that new searches are not fired when a selection happens, and you need to do something else to prompt them (is there maybe a configuration value for this?)

currently I have something like the following 
`$el.on('typeahead:change', function(e) {
        console.log(selection);
        $el.focus();
      }).on('typeahead:selected', function(e, datum) {
        selection = datum;
        $el.trigger('change');
      })`

(lots of extra code has been removed that does various things but this is the main part. 

The change event is triggered, and I get my console.log statements, but my blood hound search is never ran. If I click on the input or even navigate away from the page and navigate back the search happens (this is why I tried to do $el.focus();)

I've also tried to do 
$el.blur().focus(); //maybe it needed to 
and
var ev = $.Event('keyup');
ev.which= 13;
$el.trigger(ev);

**Comment by bryanrasmussen** [comment]
I can also do the following 
$el.trigger('typeahead:autocomplete', [e, datum]);

from within the typeahead:selected event and it comes through, but it still doesn't run a new search.

**Comment by bryanrasmussen** [comment]
anyway I figured out how to do it, but is not cleaner than my previous attempt
(also setting caret position in my code below)
var clearSelection = function($el, selection, caretpos) {
        setTimeout(function() {
          caretpos = (caretpos) ? caretpos : null;
          $el.typeahead('val', '');
          $el.trigger('blur');
          setSelection($el, selection, caretpos);
        });
      };
      var setSelection = function($el, selection, caretpos) {
        setTimeout(function() {
          $el.trigger('focus');
          $el.typeahead('val', selection);
          if (caretpos) {
            $el.get(0).setSelectionRange(caretpos, caretpos);
          }
          $el.trigger('typeahead:open');
        });
      };

called with clearSelection($el, selection, selectedItem.caretpos);

so basically the same as what I tried before but inside of setTimeout to break the event loop. 
Seems unnecessarily complicated, but I suppose this is one of the problems with universal libraries. 

I will close the issue as fixed.
